PatientWebApplication/SearchMicroserviceApi/Service/MedicineService.cs

```diff
@@ -0,0 +1,114 @@
+/***********************************************************************
+ * Module:  EquipmentService.cs
+ * Author:  Luna
+ * Purpose: Definition of the Class Service.EquipmentService
+ ***********************************************************************/
+
+using SearchMicroserviceApi.DbContextModel;
+using SearchMicroserviceApi.Model;
+using SearchMicroserviceApi.Repository;
+using System.Collections.Generic;
+
+namespace SearchMicroserviceApi.Service
+{
+    public class MedicineService : IService<Medicine>
+    {
+        public MedicineRepository medicineRepository;
+
+        public MedicineService()
+        {          
+        }
+        public MedicineService(MyDbContext context)
+        {
+            medicineRepository = new MedicineRepository(context);
+        }
+        public void New(Medicine medicine)
+        {
+            medicineRepository.New(medicine);
+        }
+
+        public void Update(Medicine medicine)
+        {
+            medicineRepository.Update(medicine);
+        }
+
+        private void deleteIfMedicinesAreEqual(Medicine firstMedicine, Medicine secondMedicine)
+        {
+            if (firstMedicine.id == secondMedicine.id)
+            {
+                medicineRepository.Delete(firstMedicine.id);
+
+            }
+        }
+
+        public void Remove(Medicine medicine)
+        {
+            List<Medicine> listOfMedicines = medicineRepository.GetAll();
+
+            foreach (Medicine medicineObject in listOfMedicines)
```

Comment:
Write this in form:
"foreach (Medicine medicineObject in medicineRepository.GetAll())"